quarkus/tests/integration/src/test/java/org/keycloak/it/cli/dist/BuildCommandDistTest.java

```diff
@@ -53,4 +54,11 @@ void failIfDevProfile(LaunchResult result) {
                 () -> "The Error Output:\n" + result.getErrorOutput() + "doesn't contains the expected string.");
         assertEquals(4, result.getErrorStream().size());
     }
+
+    @Test
+    @Launch({ "build", "--db=postgres", "--db-username=myuser", "--db-password=mypassword", "--http-enabled=true" })
+    void testFailRuntimeOptions(LaunchResult result) {
+        CLIResult cliResult = (CLIResult) result;
+        cliResult.assertError("Unknown options: '--db-username=myuser', '--db-password=mypassword', '--http-enabled=true'");
```

Comment:
I think we should also mask the password here. Could again leak into logs when ruunning it with wrong config in a ci build due to missing knowledge. now the value, which was taken from a secure place, is in those logs. so please re-add the masking

Reply:
Makes sense but we don't really sanitize this specific output when printing out invalid options.

We should fix this in a separate PR as a follow to https://github.com/keycloak/keycloak/pull/8944. We missed this output when doing that PR.

We should also figure out what to do when you wrongly type the name of a "masked" option, which should lead to the same problem.

That said, I would enhance this in a separate issue.

Reply:
One thing here though. The value is not yet "taken from a secure place" but the raw value passed to the CLI. That message happens in an early stage, prior to resolving values from config sources.

Reply:
You are right. Created https://github.com/keycloak/keycloak/issues/9720 - not sure we can/should really do something about mistyped keys, though, at least not without exceptionally adding complexity to check for blurriness in keys ;)